packages/core/src/tile/model/image/imageContent.ts

```diff
@@ -0,0 +1,196 @@
+  Constants,
+  CreatePlane,
+  Mesh,
+  Plane,
+  RawTexture2DArray,
+  Scene,
+  ShaderMaterial,
+  Texture,
+  UniformBuffer
+} from '@babylonjs/core';
+import { TypedArray } from '../../types';
+import { Datatype } from '@tiledb-inc/tiledb-cloud/lib/v2';
+import {
+  ImageShaderMaterial,
+  ImageShaderMaterialWebGPU
+} from '../../materials/imageShaderMaterial';
+import { TileContent, TileUpdateOptions } from '../tileContent';
+import { ImageDataContent } from '../../../types';
+import { Tile } from '../tile';
+
+export type ImageUpdateOptions = TileUpdateOptions & {
+  UBO?: UniformBuffer;
+  data?: {
+    raw: TypedArray;
+    width: number;
+    height: number;
+    depth: number;
+    dtype: Datatype;
+    channelLimit: number;
+  };
+};
+
+const textureOptions: Map<
+  Datatype,
+  {
+    format: number;
+    filtering: number;
+    type: number;
+    samplerType: string;
+    webGPUSampleType: string;
+  }
+> = new Map([
+  [
+    Datatype.Uint8,
+    {
+      format: Constants.TEXTUREFORMAT_RED_INTEGER,
+      type: Constants.TEXTURETYPE_UNSIGNED_BYTE,
+      filtering: Texture.NEAREST_SAMPLINGMODE,
+      samplerType: 'usampler2DArray',
+      webGPUSampleType: 'u32'
+    }
+  ],
+  [
+    Datatype.Int8,
+    {
+      format: Constants.TEXTUREFORMAT_RED_INTEGER,
+      type: Constants.TEXTURETYPE_BYTE,
+      filtering: Texture.NEAREST_SAMPLINGMODE,
+      samplerType: 'isampler2DArray',
+      webGPUSampleType: 'i32'
+    }
+  ],
+  [
+    Datatype.Uint16,
+    {
+      format: Constants.TEXTUREFORMAT_RED_INTEGER,
+      type: Constants.TEXTURETYPE_UNSIGNED_SHORT,
+      filtering: Texture.NEAREST_SAMPLINGMODE,
+      samplerType: 'usampler2DArray',
+      webGPUSampleType: 'u32'
+    }
+  ],
+  [
+    Datatype.Float32,
+    {
+      format: Constants.TEXTUREFORMAT_RED,
+      type: Constants.TEXTURETYPE_FLOAT,
+      filtering: Texture.NEAREST_SAMPLINGMODE,
+      samplerType: 'sampler2DArray',
+      webGPUSampleType: 'f32'
+    }
+  ]
+]);
+
+export class ImageContent extends TileContent {
+  private texture?: RawTexture2DArray;
+  private material?: ShaderMaterial;
+
+  constructor(scene: Scene, tile: Tile<ImageDataContent, ImageContent>) {
+    super(scene, tile);
+
+    //Create plane mesh for the image tile
+    const mesh = CreatePlane(
+      tile.id.toString(),
+      {
+        width: 2 * tile.boundingInfo.boundingBox.extendSize.x,
+        height: 2 * tile.boundingInfo.boundingBox.extendSize.z,
+        sideOrientation: Mesh.DOUBLESIDE,
+        sourcePlane: new Plane(0, 1, 0, 0)
+      },
+      scene
+    );
+
+    // Hide image tile untill data are loaded
+    // Visibility should be restored once the data are loaded
+    mesh.visibility = 0;
+    mesh.isPickable = false;
+    mesh.position = tile.boundingInfo.boundingBox.center;
+
+    // The it the top level image tile so add both layer mask to be renderered by all cameras
+    mesh.layerMask = tile.mask;
+
+    // Remove unwanted rotations added by default
```

Comment:
who adds these by default? what happens when their behavior changes?

Reply:
These are added [here](https://github.com/BabylonJS/Babylon.js/blob/ef8946fdae95f2515cf1c0b1560ba63d3ccdbdcb/packages/dev/loaders/src/glTF/2.0/glTFLoader.ts#L592). In the rare case where these change there shouldn't be any difference in our case since we already remove them.